fifo_test.go

```diff
@@ -42,15 +42,16 @@ func TestFifoCancel(t *testing.T) {
 		leakCheckWg = nil
 	}()
 
-	_, err = OpenFifo(context.Background(), filepath.Join(tmpdir, "f0"), syscall.O_RDONLY|syscall.O_NONBLOCK, 0600)
+	f, err := OpenFifo(context.Background(), filepath.Join(tmpdir, "f0"), syscall.O_RDONLY|syscall.O_NONBLOCK, 0600)
+	assert.Exactly(t, nil, f)
 	assert.NotNil(t, err)
 
 	assert.NoError(t, checkWgDone(leakCheckWg))
 
 	ctx, cancel := context.WithTimeout(context.Background(), 500*time.Millisecond)
 	defer cancel()
 
-	f, err := OpenFifo(ctx, filepath.Join(tmpdir, "f0"), syscall.O_RDONLY|syscall.O_CREAT|syscall.O_NONBLOCK, 0600)
+	f, err = OpenFifo(ctx, filepath.Join(tmpdir, "f0"), syscall.O_RDONLY|syscall.O_CREAT|syscall.O_NONBLOCK, 0600)
```

Comment:
Oh! Not a blocker (at all), but if you want to reduce code-churn, perhaps reformat the octal values to the new format (`0o600`) while updating these lines